psc-package/Main.hs

```diff
@@ -233,6 +254,73 @@ exec exeName = do
         (map pathToTextUnsafe ("src" </> "**" </> "*.purs" : paths))
         empty
 
+checkForUpdates :: Bool -> IO ()
+checkForUpdates applyMinorUpdates = do
+    pkg <- readPackageFile
+    db <- readPackageSet pkg
+
+    echo ("Checking " <> pack (show (Map.size db)) <> " packages for updates.")
+    echo "Warning: this could take some time!"
+
+    newDb <- Map.fromList <$> (for (Map.toList db) $ \(name, p@PackageInfo{ repo, version }) -> do
+      echo ("Checking package " <> name)
+      tagLines <- Turtle.fold (listRemoteTags repo) Foldl.list
+      let tags = mapMaybe parseTag tagLines
+      newVersion <- case parseVersion version of
+        Just parts -> do
+          when (any (isMajorReleaseFrom parts) tags) $
+            echo ("New major release available")
+          case filter (isMinorReleaseFrom parts) tags of
+            [] -> pure version
+            minorReleases -> do
+              echo ("New minor release available")
+              case applyMinorUpdates of
+                True -> do
+                  let latestMinorRelease = maximum minorReleases
+                  pure ("v" <> T.intercalate "." (map (pack . show) latestMinorRelease))
+                False -> pure version
+        _ -> do
+          echo "Unable to parse version string"
+          pure version
+      pure (name, p { version = newVersion }))
+
+    when applyMinorUpdates (writePackageSet pkg newDb)
+  where
+    parseTag :: Text -> Maybe [Int]
+    parseTag line =
+      case T.splitOn "\t" line of
+        [_sha, ref] ->
+          case T.stripPrefix "refs/tags/" ref of
+            Just tag ->
+              case parseVersion tag of
+                Just parts -> pure parts
+                _ -> Nothing
+            _ -> Nothing
+        _ -> Nothing
+
+    parseVersion :: Text -> Maybe [Int]
+    parseVersion ref =
+      case T.stripPrefix "v" ref of
```

Comment:
Looks good. I think it will be great improvement to have a one-to-one correspondence between versions and expected forms of tags (in fact I probably would have done it that way in the first place if I'd thought about it a bit more).

I think we should also remove the `pkgVersionTag` from the definition of `Package` in `Docs.Types`, but continue adding the field in the `ToJSON` instance for now for backwards compatibility. We can add a TODO comment to remove it the next time we break the JSON format.